components/addEditSale.js

```diff
@@ -25,12 +25,17 @@ const AddEditSale = ({ sale, type = null }) => {
   const [_laborCharge, setLaborCharge] = useState(0);
   const [selectedProducts, setSelectedProducts] = useState([]);
   const [updatedProducts, setUpdatedProducts] = useState([]);
-  const { inventory, error, isLoading } = useInventoryAttributes(["itemName", "id", "onHand", "companyId"]);
+  const { inventory, error, isLoading } = useInventoryAttributes(["itemName", "id", "onHand", "companyId"], type);
   const [form] = Form.useForm();
 
   useEffect(() => {
-    setUpdatedProducts(inventory);
-  }, [inventory]);
+    if (type === "view") {
```

Comment:
create a const for page type like 
const PAGE_TYPE_VIEW = "VIEW"
```suggestion
    if (type === PAGE_TYPE_VIEW) {
```